xcell/cls/cl.py

```diff
@@ -274,15 +323,58 @@ def compute_tracer_ccl(self, tr, mapper):
                 bz = A*((1.+z)/(1.+z0))**eta*0.0139/0.013872474
                 ia_bias = (z, bz)
             # Get tracer
-            return ccl.WeakLensingTracer(self.cosmo, dndz=(z, pz),
-                                         ia_bias=ia_bias)
+            ccl_tr = ccl.WeakLensingTracer(self.cosmo, dndz=(z, pz),
+                                           ia_bias=ia_bias)
         elif dtype == 'cmb_convergence':
             # TODO: correct z_source
-            return ccl.CMBLensingTracer(self.cosmo, z_source=1100)
+            ccl_tr = ccl.CMBLensingTracer(self.cosmo, z_source=1100)
         else:
             raise ValueError('Type of tracer not recognized. It can be \
                              galaxy_density, galaxy_shear or \
                              cmb_convergence!')
+        return {'name': tr,
+                'ccl_tr': ccl_tr,
+                'ccl_pr': ccl_pr,
+                'ccl_pr_2pt': ccl_pr_2pt,
+                'with_hm': with_hm,
+                'normed': normed_profile}
+
+    def _get_ccl_pk(self, ccl_tr1, ccl_tr2):
+        if ccl_tr1['with_hm'] or ccl_tr2['with_hm']:
+            if ccl_tr1['name'] == ccl_tr2['name']:
+                pr2pt = ccl_tr1['ccl_pr_2pt']
```

Comment:
This would be `None`, at this moment (see L. 298)

Reply:
nope!

Reply:
Now I see!